### PR TITLE
Decouple navigation from UI composables

### PR DESCRIPTION
**Pull request by Skrilltrax**
Navigation library recommends to decouple the navigation from the composables. They specially points to pass lambdas instead of passing the NavController directly.

https://developer.android.com/jetpack/compose/navigation#testing

### CODE REVIEW

**Comment by msfjarvis** [comment]
New rule: if you refactor to make testing easy you gotta add a test

**Comment by Skrilltrax** [comment]
Well, I can write one tomorrow 👀

File: app/src/main/java/dev/msfjarvis/lobsters/ui/navigation/Destination.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ enum class Destination(
     val startDestination = Hottest
 
     fun getDestinationFromRoute(route: String): Destination {
-      return values().firstOrNull { it.route == route } ?: error("Incorrect route password")
+      return values().firstOrNull { it.route == route } ?: error("Incorrect route passed")
```
Comment (msfjarvis) [inline app/src/main/java/dev/msfjarvis/lobsters/ui/navigation/Destination.kt:23]: I clearly need to stop working on APS for a while

**Comment by msfjarvis** [comment]
Rebased on develop

**Comment by msfjarvis** [comment]
r=me once tests are here. You can pick a3d3942e6d4fe1681884518a12f5a344d54fed2d, 74e8ce4b3994c076d8fb16878312391dfb2d0a52 and 3921539b0f0a04be0f99c5dafa1c25705949067d from #100 to get things set up.

bors d+

**Review by msfjarvis — APPROVED** [review]
bors r+

**Pull request merged**